src/test/java/teammates/test/cases/ui/pagedata/InstructorCourseStudentDetailsPageDataTest.java

```diff
@@ -0,0 +1,168 @@
+package teammates.test.cases.ui.pagedata;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
+import static org.testng.AssertJUnit.assertTrue;
+import static org.testng.AssertJUnit.assertFalse;
+
+import java.util.Arrays;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.AccountAttributes;
+import teammates.common.datatransfer.StudentAttributes;
+import teammates.common.datatransfer.StudentProfileAttributes;
+import teammates.common.util.Const;
+import teammates.test.cases.BaseTestCase;
+import teammates.ui.controller.InstructorCourseStudentDetailsPageData;
+import teammates.ui.template.StudentInfoTable;
+import teammates.ui.template.StudentProfile;
+
+public class InstructorCourseStudentDetailsPageDataTest extends BaseTestCase {
+    private String[] USERS_COMMENT_BOX_SHOWN_TO = {"student", "team", "section"};
+    
+    private StudentAttributes inputStudent;
+    private StudentProfileAttributes inputStudentProfile;
+    private String pictureUrl;
+    private boolean isAbleToAddComment;
+    private boolean hasSection;
+    private String commentRecipient;
+    
+    
+    @BeforeClass
+    public static void classSetUp() {
+        printTestClassHeader();
+    }
+    
+    @Test
+    public void allTests() {
+        ______TS("With picture key, no comment recipient");
+        String pictureKey = "examplePictureKey";
+        createStudentData(pictureKey);
+        InstructorCourseStudentDetailsPageData data = createData();
+        testData(data);
+        
+        ______TS("With empty picture key, no comment recipient");
+        pictureKey = "";
+        createStudentData(pictureKey);
+        data = createData();
+        testData(data);
+        
+        ______TS("With null picture key, no comment recipient");
+        pictureKey = null;
+        createStudentData(pictureKey);
+        data = createData();
+        testData(data);
+        
+        ______TS("With comment recipient unauthorised to see comment box");
+        data = createData("someOtherCommentRecipient");
+        testData(data);
+        
+        ______TS("With comment recipient authorised to see comment box");
+        for (String user : USERS_COMMENT_BOX_SHOWN_TO) {
+            data = createData(user);
+            testData(data);
+        }
```

Comment:
The alternative would be to duplicate that test 3 times, but it offers no real benefit.
While the comment recipient does differ in each run, the test is checking is it is a "valid" value, rather than specifically whether it's a "student", a "team", or a "section".
Thus, I believe they should be treated as a group instead of as individuals.